patches/server/0047-Add-depth-limit-to-Component-deserializer.patch

```diff
@@ -0,0 +1,127 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Luna <lunahatesgogle@gmail.com>
+Date: Wed, 31 May 2023 18:14:00 -0300
+Subject: [PATCH] Add depth limit to Component deserializer
+
+
+diff --git a/src/main/java/net/minecraft/network/chat/Component.java b/src/main/java/net/minecraft/network/chat/Component.java
+index 9c2c22ee548ad77f0912698f33de5f467f32fb7f..9d46804db18cd2d6bc8d5f82ca3eb302a4dcb2ad 100644
+--- a/src/main/java/net/minecraft/network/chat/Component.java
++++ b/src/main/java/net/minecraft/network/chat/Component.java
+@@ -44,6 +44,7 @@ import net.minecraft.util.GsonHelper;
+ import net.minecraft.util.LowerCaseEnumTypeAdapterFactory;
+ // CraftBukkit start
+ import com.google.common.collect.Streams;
++import java.util.regex.Pattern; // Scissors
+ import java.util.stream.Stream;
+ // CraftBukkit end
+ 
+@@ -254,10 +255,16 @@ public interface Component extends Message, FormattedText, Iterable<Component> {
+                 throw new IllegalStateException("Couldn't get field 'lineStart' for JsonReader", nosuchfieldexception);
+             }
+         });
++        private static final Pattern PLACEHOLDER_PATTERN = Pattern.compile("%[0-9]+\\$s"); // Scissors
+ 
+         public Serializer() {}
+ 
+-        public MutableComponent deserialize(JsonElement jsonelement, Type type, JsonDeserializationContext jsondeserializationcontext) throws JsonParseException {
++        // Scissors start
++        private MutableComponent deserialize(JsonElement jsonelement, JsonDeserializationContext jsondeserializationcontext, int depth) throws JsonParseException {
++            if (depth > 128) {
```

Comment:
I feel like this would be too high. If possible, could we make this customizable in the Scissors configuration so that if new variants come out, server owners could adjust it in real-time?

Reply:
Done